File: app/src/main/java/com/jcr/sharedtasks/model/Project.java
package com.jcr.sharedtasks.model;

import androidx.annotation.NonNull;

import java.util.List;

public class Project {

    private String projectUUID;
    private List<Task> tasks;
    private String name;

    public Project(@NonNull String projectUUID, String name) {
        this.projectUUID = projectUUID;
        this.name = name;
    }

    public Project() {
    }

    @NonNull
    public String getProjectUUID() {
        return projectUUID;
    }

    public void setProjectUUID(@NonNull String projectUUID) {
        this.projectUUID = projectUUID;
    }

    public List<Task> getTasks() {
        return tasks;
    }

    public void setTasks(List<Task> tasks) {
        this.tasks = tasks;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: app/src/main/java/com/jcr/sharedtasks/model/Project.kt
package com.jcr.sharedtasks.model

data class Project(
        var projectUUID: String,
        var tasks: List<Task>? = null,
        var name: String
) {

    constructor(projectUUID: String, name: String) : this(
            projectUUID,
            null,
            name
    )

    constructor() : this(
            "",
            ""
    )
}
